runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineTranslator.java

```diff
@@ -959,6 +961,18 @@ public void translate(
           private <InputT, OutputT> void translateMultiHelper(
               ParDo.BoundMulti<InputT, OutputT> transform,
               TranslationContext context) {
+
+            DoFn<InputT, OutputT> doFn = transform.getNewFn();
+            if (DoFnSignatures.getSignature(doFn.getClass()).stateDeclarations().size() > 0) {
```

Comment:
Might be nicer if there was a "DoFnSignatures.getSignature(...).isStateful()" or something to easily categorize these?